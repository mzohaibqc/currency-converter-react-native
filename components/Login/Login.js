import React, { useCallback, useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { faUser, faLock, faSpinner } from '@fortawesome/free-solid-svg-icons';

import Screen from 'components/Screen';
import AppLogo from 'components/AppLogo';
import InputsContainer from 'components/InputsContainer';
import { LoginButton, FormInput } from 'components/StyledComponents';
import useScreenTransform from 'hooks/useScreenTransform';
import selectors from 'selectors';
import { login as loginAction } from 'actions/app.actions';

function Login({}) {
  const [username, setUsername] = useState('');
  const [password, setPassword] = useState('');
  const theme = useSelector(selectors.getTheme);
  const loading = useSelector(selectors.getLoginLoading);
  const dispatch = useDispatch();

  // Fake login
  const login = useCallback((username, password) => {
    if (username === '' || password === '' || loading) return;
    dispatch(loginAction.pending(username, password));
  }, []);

  const { imageY, inputContainerY, imageScale } = useScreenTransform();

  return (
    <Screen backgroundColor={theme.color}>
      <AppLogo scale={imageScale} translateY={imageY} />
      <InputsContainer translateY={inputContainerY}>
        <FormInput
          inputProps={{
            placeholder: 'Username',
            keyboardType: 'email-address',
            value: username,
            onChangeText: (value) => setUsername(value),
          }}
          theme={theme}
          icon={faUser}
        />
        <FormInput
          inputProps={{
            placeholder: 'Password',
            secureTextEntry: true,
            value: password,
            onChangeText: (value) => setPassword(value),
          }}
          theme={theme}
          icon={faLock}
        />

        <LoginButton
          onPress={() => login(username, password)}
          text="Login"
          loading={loading}
          color={theme.color}
        />
      </InputsContainer>
    </Screen>
  );
}

export default Login;


