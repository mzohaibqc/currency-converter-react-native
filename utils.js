export const getAsyncActionType = (name) => {
  return {
    PENDING: `${name}: PENDING`,
    SUCCESS: `${name}: SUCCESS`,
    ERROR: `${name}: ERROR`,
    toString: () => name,
  };
};

export const sleep = n => new Promise(res => setTimeout(res, n));
