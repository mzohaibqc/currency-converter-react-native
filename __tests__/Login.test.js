import React from 'react';
import { cleanup } from 'react-native-testing-library';
import 'jest-styled-components';
import { customRender } from './utils';

import Login from 'components/Login/Login';

afterEach(cleanup);



describe('<Login />', () => {  
  it('should render', () => {
    const { getByText, baseElement } = customRender(
      <Login />
    );
    expect(baseElement).toMatchSnapshot()
  });
});
