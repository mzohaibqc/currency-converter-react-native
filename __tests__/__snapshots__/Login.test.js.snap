// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> should render 1`] = `undefined`;
